packages/components/src/List/Toolbar/DisplayModeToggle/DisplayModeToggle.component.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { ActionIconToggle } from '../../../Actions';
+import getDefaultT from '../../../translate';
+
+import theme from './DisplayModeToggle.scss';
+
+const options = ['table', 'large'];
+function getLabel(selected, t) {
+	switch (selected) {
+		case 'table':
+			return t('LIST_SELECT_DISPLAY_MODE_TABLE', { defaultValue: 'Table' });
+		case 'large':
+			return t('LIST_SELECT_DISPLAY_MODE_LARGE', { defaultValue: 'Expanded' });
+		default:
+			return t('LIST_SELECT_DISPLAY_MODE_TABLE', { defaultValue: 'Table' });
+	}
+}
+
+function DisplayModeToggle({ id, displayModes, onChange, mode, t }) {
+	const [active, setActive] = useState(mode);
```

Comment:
Your use of hook here duplicate the state. In the List component, everything is pure. We should stick with this behavior.
If a project controls the state externally, your state is desynchronized.

In the List composition mode there will be a controlled (pure) and uncontrolled (with internal state) mode. But for this List legacy, let's not change the feature.